src/plugins/data/public/search/fetch/handle_warnings.tsx

```diff
@@ -86,12 +90,20 @@ export function handleWarnings(
 /**
  * @internal
  */
-export function filterWarnings(warnings: SearchResponseWarning[], cb: WarningHandlerCallback) {
+export function filterWarnings(
+  warnings: SearchResponseWarning[],
+  cb: WarningHandlerCallback,
+  request: SearchRequest,
+  response: estypes.SearchResponse
+) {
   const unfiltered: SearchResponseWarning[] = [];
 
   // use the consumer's callback as a filter to receive warnings to handle on our side
   warnings.forEach((warning) => {
-    const consumerHandled = cb?.(warning);
+    const consumerHandled = cb?.(warning, () => ({
+      request,
```

Comment:
Note: If do not wrap it in a function, then an error occurs on the UI for some visualizations!